src/AiAdapter.js

```diff
@@ -496,15 +488,15 @@ var ticTacToe = ticTacToe || {};
    * Determines whether a square is empty.
    * @param {integer} square
    */
-  app.AiAdapter.prototype.squareIsEmpty = function (square) {
-    return (square === 0) ? true : false;
+  app.AiAdapter.prototype.isSquareEmpty = function (square, board) {
```

Comment:
I think the engine should handle this. The engine should make it so that a 1 or 2 cannot be changed. Only 0s are mutable indices.